modules/oe_content_persistent/src/ContentUuidResolver.php

```diff
@@ -0,0 +1,224 @@
+<?php
+
+declare(strict_types = 1);
+
+namespace Drupal\oe_content_persistent;
+
+use Drupal\Core\Cache\CacheBackendInterface;
+use Drupal\Core\Cache\RefinableCacheableDependencyTrait;
+use Drupal\Core\CacheDecorator\CacheDecoratorInterface;
+use Drupal\Core\Entity\EntityInterface;
+use Drupal\Core\Entity\EntityTypeManagerInterface;
+use Drupal\Core\Language\LanguageInterface;
+use Drupal\Core\Language\LanguageManagerInterface;
+use Drupal\Core\Path\AliasManagerInterface;
+use Drupal\Core\Url;
+
+/**
+ * Default implementation of Content UUID resolver.
+ */
+class ContentUuidResolver implements ContentUuidResolverInterface, CacheDecoratorInterface {
+
+  use RefinableCacheableDependencyTrait;
+
+  /**
+   * Holds the map of uuid lookups per language.
+   *
+   * @var array
+   */
+  protected $lookupMap = [];
+
+  /**
+   * The cache key to use when caching paths.
+   *
+   * @var string
+   */
+  protected $cacheKey;
+
+  /**
+   * The cache tags to use when caching paths.
+   *
+   * @var array
+   */
+  protected $uuidCacheTags = [];
+
+  /**
+   * Whether the cache needs to be written.
+   *
+   * @var bool
+   */
+  protected $cacheNeedsWriting = FALSE;
+
+  /**
+   * The entity type manager.
+   *
+   * @var \Drupal\Core\Entity\EntityTypeManagerInterface
+   */
+  protected $entityTypeManager;
+
+  /**
+   * The language manager.
+   *
+   * @var \Drupal\Core\Language\LanguageManagerInterface
+   */
+  protected $languageManager;
+
+  /**
+   * The alias manager that caches alias lookups based on the request.
+   *
+   * @var \Drupal\Core\Path\AliasManagerInterface
+   */
+  protected $aliasManager;
+
+  /**
+   * The cache backend.
+   *
+   * @var \Drupal\Core\Cache\CacheBackendInterface
+   */
+  protected $cache;
+
+  /**
+   * List of supported storages.
+   *
+   * @var array
+   */
+  protected $supportedStorages;
+
+  /**
+   * Constructs a ContentUuidResolver.
+   *
+   * @param \Drupal\Core\Entity\EntityTypeManagerInterface $entity_type_manager
+   *   The entity type manager.
+   * @param \Drupal\Core\Language\LanguageManagerInterface $language_manager
+   *   The language manager.
+   * @param \Drupal\Core\Path\AliasManagerInterface $alias_manager
+   *   The alias manager.
+   * @param \Drupal\Core\Cache\CacheBackendInterface $cache
+   *   The cache backend.
+   * @param array $supported_storages
+   *   List of supported storages.
+   */
+  public function __construct(EntityTypeManagerInterface $entity_type_manager, LanguageManagerInterface $language_manager, AliasManagerInterface $alias_manager, CacheBackendInterface $cache, array $supported_storages = []) {
+    $this->entityTypeManager = $entity_type_manager;
+    $this->languageManager = $language_manager;
+    $this->aliasManager = $alias_manager;
+    $this->cache = $cache;
+    $this->supportedStorages = $supported_storages;
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public function setCacheKey($key) {
+    $this->cacheKey = $this->getCachePrefixKey() . $key;
```

Comment:
This should not be possible on a service. Better to pass in the cachekey in all methods that need it as an argument.

Reply:
Anyway for now we have decided to remove caching.